usvm-core/src/main/kotlin/org/usvm/StateForker.kt

```diff
@@ -11,7 +12,39 @@ private typealias StateToCheck = Boolean
 private const val ForkedState = true
 private const val OriginalState = false
 
-sealed interface StateForker {
+interface StateForker {
```

Comment:
We don't need a separate TsForker. Seems like the only difference is that it works with UJoinedExpression, but it is a common API, and can be extracted into the original forker